File: Server/Models/Movie.cs

using System;
using System.Collections.Generic;

namespace BestOfTheWorst.Server.Models
{
    public class Movie
    {
        public long Id { get; set; }

        public string Title { get; set; }

        public string Overview { get; set; }

        public string Synopsis { get; set; }

        public DateTime? ReleaseDate { get; set; }

        public int Runtime { get; set; }

        public Guid? ImageId { get; set; }

        public Image Image { get; set; }

        public long EpisodeId { get; set; }

        public Episode Episode { get; set; }

        public TimeSpan Timestamp { get; set; }

        public IList<Tag> Tags { get; set; } = new List<Tag>();

        public IList<Link> Links { get; set; } = new List<Link>();

        public IList<Genre> Genres { get; set; } = new List<Genre>();
    }
}

File: ClientApp/src/app/movie.ts
import { Link } from "./link";
import { Episode } from "./episode";
import { Genre } from "./genre";

export class Image {
    id: string;
    imageUrls: {[key: number]: string};
}

export class Tag {
    id: number;
    name: string;
}

export class Movie {
    id: number;
    title: string;
    overview: string;
    releaseDate: string;
    synopsis: string;
    runtime: number;
    image?: Image;
    episode?: Episode;
    tags: Array<Tag> = [];
    links: Array<Link> = [];
    genres: Array<Genre> = [];
}

export class MovieForm {
    title: string;
    overview: string;
    synopsis: string;
    releaseDate: string;
    runtime: number;
    image: string;
    episodeId: number;
    tags: string[] = [];
    links: Link[] = [];
    genres: Genre[] = [];

    public constructor(init?: Partial<MovieForm>) {
        Object.assign(this, init);
    }
}

export class PageInfo {
    pageIndex: number;
    pageSize: number;
    totalPages: number;
    totalItems: number;
}

export class MovieList {
    paging: PageInfo;
    items: Movie[];
}